src/ua_parser/__init__.py

```diff
@@ -1 +1,83 @@
-VERSION = (0, 16, 1)
+"""The package provides top-level helpers which use a lazily initialised
+default parser. These are convenience functions, for more control it
+is perfectly acceptable to instantiate and call parsers directly.
+
+The default parser does use a cache keyed on the user-agent string,
+but its exact behaviour is unspecified, if you require a consistent
+behaviour or specific algorithm, set up your own parser (global or
+not).
+
+For convenience, direct aliases are also provided for:
+
+- :mod:`core types <.types>`
+- :mod:`caching utilities <.caching>`
+- :mod:`ua_parser.basic.Parser` as :class:`BasicParser`
+
+This way importing anything but the top-level package should not be
+necessary unless you want to *implement* a parser.
+"""
+VERSION = (1, 0, 0)
+
+from typing import Optional
+from .core import *
+from .basic import Parser as BasicParser
+from .caching import CachingParser, Clearing, LRU
+from .loaders import load_builtins, load_data, load_yaml
+
+
+parser: Parser
```

Comment:
Maybe this should be a threadlocal or contextvar? This makes them inherently thread-safe, although it means higher cache memory consumption (because there's one cache per context) and lower hit rate until steady state.

Maybe there should just be a threadlocal parser option which can be configured here, with lazy init? This way lib user could pick their poison. Especially if the caching parser is optionally thread safe, threaded user could pick between a shared thread-safe cache and threadlocal non-thread-safe caches (likely to the same underlying parser).